core/src/main/java/com/graphhopper/routing/ev/StringEncodedValue.java

```diff
@@ -30,7 +27,7 @@ public StringEncodedValue(String name, int expectedValueCount, boolean storeTwoD
 
         this.maxValues = roundUp(expectedValueCount);
         this.values = new ArrayList<>(maxValues);
-        this.indexMap = new ObjectIntHashMap<>(maxValues);
+        this.indexMap = new HashMap<>(maxValues);
```

Comment:
Could we avoid this with a dedicated getter instead of using field access? I'm not sure if we want to deal with auto(un)boxing in a potentially hot code path.

Reply:
Is this really about field access. I thought we need a custom serializer for hppc, but I might be wrong. I thought about the auto unboxing for a moment but then thought it's probably not as bad bc the maps will be very small. But you are more than welcome to make this work with the hppc map again. I just did not feel like spending a lot more time on this, also because currently the default GraphHopper server does not use StringEncodedValue currently.

Reply:
I tried changing `indexMap` to the concrete type `ObjectIntHashMap` and got another error which is due to the fact that `HashOrderMixingStrategy` is an abstract type and Jackson does not know which type to use upon de-serialization.

Reply:
It works when we add `jackson-datatype-hppc`: https://github.com/FasterXML/jackson-datatypes-collections
I tried this here including a little speed test: fa80be946c6620290c680a72f41748ff4de0fd3b
I could not measure a notable difference, though so I would go with java.util for now.